src/transformers/models/gpt_sw3/tokenization_gpt_sw3.py

```diff
@@ -127,15 +127,15 @@ def __init__(
         name_or_path = kwargs.get("name_or_path")
         if name_or_path is None:
             logger.warning(
-                "name_or_path not provided, will work for all GPTSw3 models except gpt-sw3-7b,"
+                "name_or_path not provided, will work for all GPTSw3 models except gpt-sw3-6.7b,"
                 " you are testing the model, this can safely be ignored"
             )
             name_or_path = "None"
 
         # Default definitions for our 2 tokenizer versions, with None-checks to enable proper testing
         eos_token = "<|endoftext|>" if eos_token is None else eos_token
         unk_token = "<unk>" if unk_token is None else unk_token
-        if "gpt-sw3-7b" in name_or_path:
+        if ("gpt-sw3-7b" in name_or_path) or ("gpt-sw3-6.7b" in name_or_path):
```

Comment:
I think this will match when `name_or_path`is 6.7b-v2, which it shouldn't. Changing the check to endswith or some regexp should solve it.

Reply:
Ah, I wasn't aware that 6.7b-v2 had a different vocabulary. I added a commit to address that.